pyproject.toml

```diff
@@ -42,12 +42,12 @@ classifiers = [
     'Framework :: Pytest',
     'Topic :: Software Development :: Libraries :: Python Modules',
 ]
-requires-python = '>=3.7'
+requires-python = '>=3.8'
 dynamic = ['version']
 dependencies = [
     'pytest>=7',
     'black>=23',
```

Comment:
i guess you can remove black as a dependency now?

Reply:
`black_format` still exists as a function for now, I want to push that as a follow up PR to keep a smaller diff. "ruff format" in this context is for our CI, not for the functionality 😂

Reply:
Actually I'll just add here as a follow-up commit.